gapis/api/api.go

```diff
@@ -106,3 +106,11 @@ func Register(api API) {
 func Find(id ID) API {
 	return apis[id]
 }
+
+// CloneContext is used to keep track of references when cloning API objects.
+type CloneContext map[interface{}]interface{}
+
+// NewCloneContext creates a new empty CloneContext.
+func NewCloneContext() CloneContext {
```

Comment:
Constructors for trivial default-initialized types are frowned upon in Go.
It would be more common to build these with: `CloneContext{}`